core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -154,15 +155,17 @@ private[spark] class HighlyCompressedMapStatus private (
 
   override def writeExternal(out: ObjectOutput): Unit = Utils.tryOrIOException {
     loc.writeExternal(out)
-    emptyBlocks.writeExternal(out)
+    markedBlocks.writeExternal(out)
```

Comment:
given the immutability of them, this should be unnecessary, right?  We can just call it once when we construct it?  (I totally agree we need to call it at least once, but I think its more appropriate in the other spot.)